app/models/concerns/pdf_visualization_operations.rb

```diff
@@ -75,7 +86,7 @@ def find_or_create_form
     end
 
     def downloadable_as_pdf?
-      xml? && form&.xsl_fo&.present?
+      (xml? && form&.xsl_fo&.present?) || (form? && message.html_visualization.present?)
```

Comment:
v klude toto rozdel na dva riadky kde prvy bude return true if xml? ... a druhy tiez return true if form? && .. 
 a potom return false. je to citatelnejsie